rudder-web/src/main/scala/com/normation/rudder/web/rest/RestExtractorService.scala

```diff
@@ -679,5 +713,27 @@ case class RestExtractorService (
     }
   }
 
+  def extractQuery (params : Map[String,List[String]]) : Box[Option[Query]] = {
+    extractOneValue(params,"query")(convertToQuery) match {
+      case Full(None) =>
+        extractOneValue(params,CRITERIA)(convertToQueryCriterion) match {
+          case Full(None) => Full(None)
+          case Full(Some(criterion)) =>
+            for {
+              optType <- extractOneValue(params,TARGET)(convertToQueryReturnType)
```

Comment:
ie: a request /rudder/api/latest/nodes?where=myCriterion&select=notagoodvalue will end in::

    {
  "action": "listAcceptedNodes",
  "result": "error",
  "errorDetails": "Query return type 'notagoodvalue' is not valid"
* Connection #0 to host localhost left intact